src/js/menu/menu-button.js

```diff
@@ -58,7 +59,7 @@ class MenuButton extends ClickableComponent {
     this.buttonPressed_ = false;
     this.el_.setAttribute('aria-expanded', 'false');
 
-    if (this.items && this.items.length === 0) {
+    if (this.items && (this.items.length === 0 || (this.items.length === 1 && this.items[0] instanceof OffTextTrackMenuItem))) {
```

Comment:
I don't this should be handled in the `MenuButton` component; it should be in the sub-class component `CaptionsButton`, using the [`update`](https://github.com/videojs/video.js/blob/master/src/js/control-bar/text-track-controls/captions-button.js#L52-L67) method which explicitly understands the different `threshold` if the OffTextTrackMenuItem is included.